gapic/src/main/com/google/gapid/perfetto/models/SliceTrack.java

```diff
@@ -67,14 +67,76 @@ protected Slice buildSlice(Row row, ArgSet args) {
 
   public static SliceTrack forGpuQueue(QueryEngine qe, GpuInfo.Queue queue) {
     return new WithQueryEngine(qe, "gpu_slice", queue.trackId) {
+      private final String GPU_COLUMNS = "render_target, render_target_name, render_pass, render_pass_name, command_buffer, command_buffer_name";
+      private final String GPU_SLICE_SQL = "select " + BASE_COLUMNS + ", " + GPU_COLUMNS + " from %s where id = %d";
+      private final String GPU_SLICES_VIEW = "select " + BASE_COLUMNS + ", " + GPU_COLUMNS + " from %s where track_id = %d";
+      private final String GPU_SLICES_SQL =
+          "select " + BASE_COLUMNS + ", " + GPU_COLUMNS + " from %s " +
+              "where ts >= %d - dur and ts <= %d order by ts";
+
+      private String gpuSliceSql(long id) {
+        return format(GPU_SLICE_SQL, tableName("slices"), id);
+      }
+
+      @Override
+      protected ListenableFuture<?> initialize() {
+        String slices = tableName("slices");
+        String window = tableName("window");
+        String span = tableName("span");
+        return qe.queries(
+            dropTable(span),
+            dropView(slices),
+            dropTable(window),
+            createWindow(window),
+            createView(slices, format(GPU_SLICES_VIEW, table, trackId)),
+            createSpan(span, window + ", " + slices + " PARTITIONED depth"));
+      }
+
+      @Override
+      protected ListenableFuture<Data> computeData(DataRequest req) {
+        Window window = Window.compute(req, 5);
+        return transformAsync(window.update(qe, tableName("window")), $ ->
+            window.quantized ? computeQuantSlices(req) : computeSlices(req));
+      }
+
+      private ListenableFuture<Data> computeSlices(DataRequest req) {
+        return transformAsync(qe.query(gpuSlicesSql(req)), res ->
+          transform(qe.getAllArgs(res.stream().mapToLong(r -> r.getLong(8))), args -> {
+            int rows = res.getNumRows();
+            Data data = new Data(req, new long[rows], new long[rows], new long[rows], new int[rows],
+                new String[rows], new String[rows], new ArgSet[rows]);
+            res.forEachRow((i, row) -> {
+              long start = row.getLong(1);
+              data.ids[i] = row.getLong(0);
+              data.starts[i] = start;
+              data.ends[i] = start + row.getLong(2);
+              data.categories[i] = row.getString(3);
+              data.titles[i] = row.getString(4);
+              data.depths[i] = row.getInt(5);
+              if (data.depths[i] == 0) {
```

Comment:
Add comment saying something like  "Append debug marker of target frame buffer if available."